src/Aspire.Hosting/Dashboard/DashboardTokenProvider.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text;
+using Aspire.Hosting.Utils;
+using Microsoft.Extensions.Options;
+
+namespace Aspire.Hosting.Dashboard;
+
+internal interface IDashboardTokenProvider
+{
+    public string BrowserToken { get; }
+    public string OltpToken { get; }
+    public string ResourceServerToken { get; }
+}
+
+internal class DashboardTokenProvider : IDashboardTokenProvider
+{
+    public DashboardTokenProvider(IOptions<DashboardAuthenticationOptions> dashboardAuthenticationOptions)
+    {
+        BrowserToken = dashboardAuthenticationOptions.Value.BrowserToken ?? GenerateToken();
+        OltpToken = GenerateToken();
+        ResourceServerToken = GenerateToken();
+    }
+
+    private static string GenerateToken()
+    {
+        var rawToken = PasswordGenerator.Generate(24, true, true, true, true, 0, 0, 0, 0);
+        var rawTokenBytes = Encoding.UTF8.GetBytes(rawToken);
+        var encodedToken = Convert.ToHexString(rawTokenBytes).ToLower();
```

Comment:
.NET 9 has `Convert.ToHexStringLower`. Maybe add a TODO here or add conditional code for `#if NET9_OR_GREATER`, etc.

https://learn.microsoft.com/en-us/dotnet/api/system.convert.tohexstringlower?view=net-9.0